silkrpc/commands/erigon_api.cpp

```diff
@@ -324,4 +324,43 @@ boost::asio::awaitable<void> ErigonRpcApi::handle_erigon_watch_the_burn(const nl
     co_return;
 }
 
+
+// https://eth.wiki/json-rpc/API#erigon_blockNumber
+boost::asio::awaitable<void> ErigonRpcApi::handle_erigon_block_number(const nlohmann::json& request, nlohmann::json& reply) {
+    auto params = request["params"];
+    std::string block_id;
+    if (params.size() == 0) {
+       block_id = core::kLatestExecutedBlockId;
+    } else if (params.size() > 1) {
+        auto error_msg = "invalid erigon_getBlockNumber params: " + params.dump();
+        SILKRPC_ERROR << error_msg << "\n";
+        reply = make_json_error(request["id"], 100, error_msg);
+        co_return;
+    } else {
+        block_id = params[0];
+    }
```

Comment:
I suggest to switch the last two branches and adjust the condition accordingly:
```
    if (params.size() == 0) {
       block_id = core::kLatestExecutedBlockId;
    } else if (params.size() == 1) {
        block_id = params[0];
    } else {
        auto error_msg = "invalid erigon_getBlockNumber params: " + params.dump();
        SILKRPC_ERROR << error_msg << "\n";
        reply = make_json_error(request["id"], 100, error_msg);
        co_return;
    }
```
It seems more natural and also separates the valid parameter values (0, 1) from the invalid ones (all the other values)

Reply:
ok